### PR TITLE
provider: upgrade Docker to v1.11

### PR DESCRIPTION
**Pull request by sirspinach**
* Install Docker 1.11 on machines instead of v1.9.

* Remove `apt-get update` optimisation. Ignoring the universe repo prevents
  successful install of Docker 1.11 dependencies.

* Make sure that Docker API calls to /container/create use nonempty HostConfig.
  Empty HostConfig worked with Docker 1.9 but fails on 1.10+.

### CODE REVIEW

**Comment by sirspinach** [comment]
Can't download 1.12 yet.
```
Provides: 
1.11.2-0~wily - 
1.11.1-0~wily - 
1.11.0-0~wily - 
1.10.3-0~wily - 
1.10.2-0~wily - 
1.10.1-0~wily - 
1.10.0-0~wily - 
1.9.1-0~wily - 
1.9.0-0~wily - 
1.8.3-0~wily - 
1.8.2-0~wily - 
1.8.1-0~wily - 
1.8.0-0~wily - 
1.7.1-0~wily - 
1.6.2-0~wily - 
1.6.1-0~wily - 
1.6.0-0~wily - 
1.5.0-0~wily -
```

**Comment by ZhongtianWang** [comment]
Commit message should be "provider: Upgrade Docker to v1.11". You forgot to capitalize "Upgrade" 😃

File: minion/docker/docker.go [hunk 1]
```diff
@@ -380,6 +380,7 @@ func (dk docker) create(name, image string, args []string,
 	container, err := dk.CreateContainer(dkc.CreateContainerOptions{
 		Name:   name,
 		Config: &dkc.Config{Image: string(image), Cmd: args, Labels: labels, Env: envList},
+               HostConfig: hc,
```
Comment (ZhongtianWang) [inline minion/docker/docker.go]: For go style, the indentation after the colon should match. Otherwise, whenever I open this file, my editor complains. 😢

**Comment by ZhongtianWang** [comment]
Done with first round review

File: cluster/provider/cloud_config.go [hunk 2]
```diff
@@ -81,16 +81,10 @@ initialize_minion() {
 }
 
 install_docker() {
-	# Disable default sources to speed up the initial 'apt-get update'
```
Comment (nlsun) [inline cluster/provider/cloud_config.go]: Why delete this?

Comment (ejj) [inline cluster/provider/cloud_config.go]: Ah Steven and I discussed this off channel.

Basically, newer versions of docker require some packages that aren't on the system by default, so we have to do an apt-get update to install anyways

File: minion/docker/docker.go (hunk 1 above)
Comment (sirspinach) [inline minion/docker/docker.go]: done

**Comment by sirspinach** [comment]
ready for more review~!

**Comment by ejj** [comment]
Merged

**Pull request closed without merging**